src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs

```diff
@@ -296,12 +310,16 @@ public static async Task PutCertificateCollectionInCluster(CertificateDefinition
                     Permissions = certDef.Permissions,
                     SecurityClearance = certDef.SecurityClearance,
                     Password = certDef.Password,
-
                 };
 
-                if (x509Certificate.HasPrivateKey)
+                if (includePrivateKey)
                 {
-                    // avoid storing the private key
+                    // here we need to keep the private key, since in pull replication we are the initiator of the communication.
+                    currentCertDef.Certificate = Convert.ToBase64String(x509Certificate.Export(X509ContentType.Pfx));
```

Comment:
This shouldn't be using the same mechanism, as certificates that we store that aren't private information.
This should be _encrypted_, talk with @iftahbe on how we store sensitive information like that.
I would also like it to be completely separated so you can't re-use existing value just by knowing its thumbprint.

Reply:
Hm.... you are correct. I didn't take into account the fact that we need this to be cross cluster.
In this case, I think we can just put the cert (with private key) in the db record and be done with it.
We already have similarly sensitive stuff with connection strings, and they go to the same location.